high_card.rb

```diff
@@ -1,5 +1,63 @@
+
 # Use these two arrays to generate a deck of cards.
 ranks = ["A", 2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K" ]
 suits = [ "hearts", "spades", "clubs", "diamonds" ]
 
-players = []
+def build_deck(ranks, suits)
+  deck = []
+  ranks.each_with_index do |rank, i|
+    suits.each do |suit|
+      card = {
+        rank: rank,
+        suit: suit,
+        value: i
+      }
+      deck << card
+    end
+  end
+  deck.shuffle
+end
+deck = build_deck(ranks, suits)
+
+def choose_players
+  players = []
+  while true do
+    puts "#{players.length} players so far.  Enter a player name, or type 'play':"
+    name = gets.chomp
+    if name == 'play'
+      break
+    end
+    players << name
+  end
+  players
+end
+players = choose_players
+
+puts "the players are #{players}"
+puts "**************************"
+
+hands = []
+
+def make_hands(players, hands, deck)
```

Comment:
What if we just dealt one card at a time?  Would that make it easier or harder?  The instructions didn't mention dealing all the cards first.